lib/controller/LocationController.js

```diff
@@ -157,7 +157,8 @@ LocationController.push({
   handler: async (request) => {
     const { s1 } = request.query;
     const features = CompositeId.decode(s1);
-    return CentrelineDAO.byFeatures(features);
+    const locations = await CentrelineDAO.byFeatures(features);
+    return locations.filter(location => location !== null);
```

Comment:
This change will affect any frontend callsites of `getLocationsByCentreline` _or_ its single-location counterpart `getLocationByCentreline`.  It also points out that `CentrelineDAO.byFeatures` can return `null` values, which we might need to check for in several places (to prevent future bugs with similar causes).

Also: is it better to filter these out, or allow `null` return values?  If the former, should we do that here, or in `CentrelineDAO.byFeatures`?  This may need a bit more thought before being merged, and possibly also some additional test cases in our DB and REST API tests to ensure whatever change we make here doesn't break something.